server/channels/app/post_metadata.go

```diff
@@ -142,6 +142,18 @@ func (a *App) PreparePostForClient(rctx request.CTX, originalPost *model.Post, o
 	a.preparePostFilesForClient(rctx, post, opts)
 
 	if post.Type == model.PostTypeBurnOnRead {
+		// For the author/sender, fill ExpireAt from TemporaryPost table in case all recipients have revealed
```

Comment:
This looks good but @pvev wondering if we can cache this. Because querying database on each occasion would be a net negative impact on getPosts APIs.

Considering this is not a blocker, would you mind if we can come up with a more performant solution?